articles/deciding-between-cloud-spanner-and-microsoft-cosmos-db/index.md

```diff
@@ -0,0 +1,55 @@
+
+Microsoft joined the cloud market and repurposed its on-premises software for the cloud. These software include [Windows Server](https://www.microsoft.com/en-us/windows-server), [SQL Server](https://www.microsoft.com/en-gb/sql-server/sql-server-downloads), [Office](https://www.office.com/), [.Net](https://dotnet.microsoft.com/), [Microsoft Dynamics 365](https://dynamics.microsoft.com/en-us/), and [SharePoint](https://support.microsoft.com/en-us/office/what-is-sharepoint-97b915e6-651b-43b2-827d-fb25777f446f). Many organizations use Windows and other Microsoft associated software, which is essential for Cosmos DB&#39;s success. Cosmos DB tightly integrates with other Microsoft applications. Thus, organizations using a lot of Microsoft software find sense to also use Cosmos DB. This builds on customer loyalty.
+
+On the flip side, users associate Cosmos DB platform with some imperfections. Cosmos DB fails to provide a seamless way to switch between [database-provisioned throughput and container-provisioned throughput](https://docs.microsoft.com/en-us/azure/cosmos-db/set-throughput). You have to recreate the database for you to switch successfully. Besides, Cosmos DB does not allow documents from different logical partition to participate in the same transaction. And writes from different containers cannot be part of the same transaction.
+
+### Pros and Cons of Google Cloud Spanner
+
+One of the strengths of Google Cloud Spanner is its strong offering in containers. The [Kubernetes standard](https://cloud.google.com/kubernetes-engine) developed by Google is now also offered by Azure. Google Cloud Platform ([GCP](https://console.cloud.google.com/)) deals with high compute offerings, such as big data, machine learning, and analytics. GCP provides significant scale and load balancing.
+
+Spanner is capable of handling large volumes of data. Its use is not limited to applications of large sizes. Further, it allows the standardization of a single database engine for all workloads requiring an RDBMS. This is very beneficial to organizations.
+
+On the downside, it may be challenging to create an instance in a local environment for Cloud Spanner. A development environment needs to be as close a match to production as possible. But this is not the case with Spanner because you must rely on a full spanner instance to accomplish this need. You may choose a single region instance to save on costs.
+
+### Major Differences between Cloud Spanner and Microsoft Cosmos DB
+
+Google Spanner supports concurrent ACID (atomicity, consistency, isolation, and durability) guarantees across more than one operation. Cosmos DB supports ACID only within the limit of a single operation. ACID are properties that ensure reliable processing of [database transactions](https://en.wikipedia.org/wiki/Database_transaction). ACID databases guarantee data correctness and maintainability in extensive production systems.
```

Comment:
There must be a link on ACID operations in the eng ed community. Link it here if you find it.

Reply:
> There must be a link on ACID operations in the eng ed community. Link it here if you find it.

I could not find an article on ACID operations in the eng-ed community. I think it should be my next topic.